js_src/dismiss.ts

```diff
@@ -0,0 +1,40 @@
+export default class Dismiss {
+    private refersToElement;
+    private doesSoftDismiss;
+
+    public static softDismissDuration = 250; // ms
+
+    constructor(public element) {
+        this.refersToElement = document.querySelector(element.getAttribute("data-dismiss"));
+
+        this.doesSoftDismiss = element.hasAttribute("data-dismiss-soft");
+        
+        element.addEventListener("click", () => { this.dismiss() });
+    }
+
+    private dismiss() {
+        // Remove immediately, if no support for animate
+        if (!this.doesSoftDismiss || !this.refersToElement.animate) {
+            this.refersToElement.remove();   
+        } else {
+            this.refersToElement.animate([
+                { opacity: 1.0 },
+                { opacity: 0 }
+            ], {
+                duration: Dismiss.softDismissDuration
+            });
+            setTimeout(() => {
+                this.refersToElement.remove();
+            }, Dismiss.softDismissDuration)
+        }
+    }
+    
+    static find(query) {
+        const elements = document.querySelectorAll(query);
+        let toggles = [];
+        for (let i = 0; i < elements.length; i++) {
```

Comment:
You can use a `forEach` here. Something like
```js
Array.from(elements).forEach(data => toggles.push(new Dismiss(data)))
```

Reply:
How about `for (const element of Array.from(elements))` cause callbacks are not so cool

Reply:
(will stay, as discussed in Discord)